compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -289,6 +281,8 @@ class PostTyper extends MacroTransform with SymTransformer  { thisTransformer =>
             case _                                  =>
           }
           super.transform(tree)
+        case Typed(Ident(nme.WILDCARD), _) =>
+          tree // skip checking pattern type
```

Comment:
Why?

Reply:
Because type bounds in a pattern need not conform to selector bounds. I.e. you have 

    type Tree[T >: Null <: Type]

You are still allowed to write

    case x: Tree[_]

(which translates to)

    case x: (_: Tree[_])

Reply:
I am going to refine & explain the logic better